### ISSUE TITLE
Error installing Symfony using Composer 1.3.x

### ISSUE THREAD

**Issue by DavidGarciaCat** [symfony/symfony#21749]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.8.16 & 2.8.17 (tested with these 2 versions)

Developing code locally using macOS Sierra & PHP 7, I usually run `composer self-udpate` before manage my dependencies. Everything worked fine, but when I deployed to Linux servers then the error appeared:

```
> Incenteev\ParameterHandler\ScriptHandler::buildParameters
Updating the "app/config/parameters.yml" file
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache

  [Symfony\Component\DependencyInjection\Exception\InvalidArgumentException]
  Unable to parse file "/.../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml".

  [InvalidArgumentException]
  The XML file "/.../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml" is not valid.

  [Symfony\Component\Debug\Exception\ContextErrorException]
  Notice: Undefined property: DOMDocument::$documentElement

Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the symfony-scripts event terminated with an exception

  [RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:

    [Symfony\Component\DependencyInjection\Exception\InvalidArgumentException]
    Unable to parse file "/.../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml".

    [InvalidArgumentException]
    The XML file "/.../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml" is not valid.

    [Symfony\Component\Debug\Exception\ContextErrorException]
    Notice: Undefined property: DOMDocument::$documentElement
```

For some reason, `Composer 1.2.4` installs Symfony and run the `symfony-scripts` as usual and without errors, but `Composer 1.3.1` and `Composer 1.3.2` are throwing an Exception (thrown by Symfony) when I run `composer install` (I insist, it happened on Linux environment, but worked fine on Mac).

I opened an issue on Composer project here https://github.com/composer/composer/issues/6203 but given their POV this issue is caused by Symfony instead of composer, even when the issue just appeared after upgrade Composer.

If this is a Symfony error, do you think you can work on a fix?

Thanks,

**Comment by jvasseur** [comment]
This looks like the same issue as https://github.com/sensiolabs/SensioDistributionBundle/issues/299 can you try updating the distribution bundle to the last version ?

**Comment by DavidGarciaCat** [comment]
Well, this is what I did:
1. Update my `composer.json` (due I am using `sensio/distribution-bundle` version `^4.0`, so now I was expecting to use `^5.0` based on https://github.com/sensiolabs/SensioDistributionBundle/issues/299 comments)
2. Remove the `_configurator` as described at [SensioDistributionBundle UPGRADE.md](https://github.com/sensiolabs/SensioDistributionBundle/blob/master/UPGRADE.md) file
3. Double check the `web/app.php` & `web/app_dev.php` (and other environments) web controllers to be sure they're set-up properly (based on [symfony/symfony-standard](https://github.com/symfony/symfony-standard) files)

After run `composer install` now I am getting this:

```
Generating autoload files
> Incenteev\ParameterHandler\ScriptHandler::buildParameters
Updating the "app/config/parameters.yml" file
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache

Fatal error: Class 'Symfony\Component\HttpKernel\Kernel' not found in /Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php on line 6

Call Stack:
    0.0004     358968   1. {main}() /Users/davidgarcia/htdocs/crowd2fund/app/console:0
    0.0081     727896   2. require_once('/Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php') /Users/davidgarcia/htdocs/crowd2fund/app/console:11

PHP Fatal error:  Class 'Symfony\Component\HttpKernel\Kernel' not found in /Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php on line 6
PHP Stack trace:
PHP   1. {main}() /Users/davidgarcia/htdocs/crowd2fund/app/console:0
PHP   2. require_once() /Users/davidgarcia/htdocs/crowd2fund/app/console:11
Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the symfony-scripts event terminated with an exception


  [RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:
  Fatal error: Class 'Symfony\Component\HttpKernel\Kernel' not found in /Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php on line 6
  Call Stack:
      0.0004     358968   1. {main}() /Users/davidgarcia/htdocs/crowd2fund/app/console:0
      0.0081     727896   2. require_once('/Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php') /Users/davidgarcia/htdocs/crowd2fund/app/console:11
  PHP Fatal error:  Class 'Symfony\Component\HttpKernel\Kernel' not found in /Users/davidgarcia/htdocs/crowd2fund/app/AppKernel.php on line 6
  PHP Stack trace:
  PHP   1. {main}() /Users/davidgarcia/htdocs/crowd2fund/app/console:0
  PHP   2. require_once() /Users/davidgarcia/htdocs/crowd2fund/app/console:11
```

So I believe I am doing something wrong... Ideas?

Thanks,

**Comment by DavidGarciaCat** [comment]
Keeping the latest `4.x` version is - apparently - working so I will run more tests. I believe we can close for now this issue. Thanks guys,

**Comment by stof** [comment]
@DavidGarciaCat you also need to update the app/console (or bin/console if you use the SF3 structure) to register the autoloader, as ``bootstrap.php.cache`` does not requrie the autoloading anymore in 5.0 (silimar to the change you did in ``web/app.php`` btw)

**Comment by DavidGarciaCat** [comment]
@stof 👍

Then looks like I need to update the `app/console` too, so I will try it.

Thanks for the clue!

**Comment by davividal** [comment]
I was having the same issue: upgrading from 2.6 -> 2.8, PHP 5.5 -> 7.1.
Updating sensio/distribution-bundle (v3.0.36 => v4.0.38) did the trick.

**Comment by lethak** [comment]
upgrading to php7, this worked for me

`composer require sensio/distribution-bundle ~4.0`

**Comment by KrzysztofMadejski** [comment]
`composer update sensio/distribution-bundle` worked for me on composer 1.6.3, php 7.3. https://github.com/composer/composer/issues/6203 mentions that it's a bug in `sensio/distribution-bundle` that was fixed only in v5.0.17.

**Comment by eilsongithub** [comment]
Dear Sirs. I have an api in cpanel (composer 1.8.4; php7.2; symphony 4) to which I am consulting with postman, however I get this error:

![image](https://user-images.githubusercontent.com/44086967/55673007-b26eea80-5867-11e9-81df-0cb04f9c735c.png)

in local I get this

![image](https://user-images.githubusercontent.com/44086967/55673026-f104a500-5867-11e9-9ee5-e3e84bfe0534.png)

Can someone give me some idea?
